packages/treasury/src/vault.js

```diff
@@ -130,19 +143,31 @@ export function makeVaultKit(
       locked: getCollateralAmount(),
       debt: runDebt,
       collateralizationRatio,
-      liquidated: !active,
+      liquidated: vaultState === VaultState.CLOSED,
+      vaultState,
     });
 
-    if (active) {
-      uiUpdater.updateState(uiState);
-    } else {
-      uiUpdater.finish(uiState);
+    switch (vaultState) {
+      case VaultState.ACTIVE:
+      case VaultState.LIQUIDATING:
+        uiUpdater.updateState(uiState);
+        break;
+      case VaultState.CLOSED:
+        uiUpdater.finish(uiState);
+        break;
+      default:
+        throw Error(`unreachable vaultState: ${vaultState}`);
     }
   }
 
   function liquidated(newDebt) {
     runDebt = newDebt;
-    active = false;
+    vaultState = VaultState.CLOSED;
```

Comment:
for a future cleanup, perhaps a state argument should just be provided to `updateUiState` to coming the notification and update?